### PR TITLE
Hypernoblium crystal rework. Turns it into a room temperature regulating crystal foam grenade, and removes its clothing pressure proofing.

### PR DESCRIPTION
**Pull request by Pickle-Coding**
## About The Pull Request
Reworks the hypernoblium crystal. Turns it into a grenade, that when detonated, will release a foam that regulates the air and floor's temperature to room temperature, and temporarily extinguishes fires for the duration. The foam lasts for 16 seconds, and has a total heat capacity of 1.755MJ. Changes crystallizer recipe temperature requirements. Minimum temperature 1,200K, maximum 1,400K.

Removes its ability to pressure proof clothes.

Drafted until I add the sprite for the active state of the hypernoblium crystal grenade.

## Why It's Good For The Game
Atmospherics progression to allow people to completely ignore a part of an atmospheric problem forever isn't very interesting. The pressure proofing just lets people ignore an interesting mechanic (pressure damage from low pressure) and is kinda powergamey.

Changing the purpose of the crystal to make it much easier to regulate the room's temperature and fire suppression is a more engaging and less powercreep crystal than one that just removes a mechanic for a player. Cooling or heating the room with the foam and slowing or suppressing fires is a lot more fun too.

## Changelog

:cl:
add: Hypernoblium crystal is now a grenade. When detonated, it will spew foam that has exceptional ability to regulate the room's temperature, and can suppress fires.
remove: You can no longer pressure proof your clothes with hypernoblium crystals.
balance: Changes hypernoblium crystal crystallizer recipe minimum temperature to 1,200K and maximum temperature to 1,400K.
/:cl:

### CODE REVIEW

**Comment by Pickle-Coding** [comment]
Forgot my changelog. Should add balance and feature labels.

**Comment by Pickle-Coding** [comment]
Unintended effect, but since reinforced floors have a heat capacity of e30 or something, the foam is much more powerful when used on reinforced floors since they conduct from floors quickly.

**Comment by CrimCroogle** [comment]
Okay but why.
we already have healium crystals

**Comment by trexter5555** [comment]
you can say the same thing "completely able to ignore a problem of atmospherics" about a lot of currently existing things such as most modsuits or atmos firesuits, which are available roundstart, and I don't think you really understand what "powergaming" means here. i'm not an atmos expert or anything but having pressure proof potions from something that you have to work for seems fine, and if you're going to insist that it's unbalanced and powergamey, why not remove xenobio fireproof potions, or even the pressureproof potions you can still make in xenobio which is significantly easier to understand than atmos. remove all roundstart spacesuits because it lets you permanently ignore a part of atmospherics. remove roundstart oxygen, it lets you ignore a part of atmospherics. i just dont understand your argument for this other than you dont like it so you're removing it.

**Comment by SylvaraWasTaken** [comment]
Why would you replace an existing feature in exchange for a feature that is already covered by about three different mechanics that do it faster and cheaper? What's even the point of this PR besides a thinly-veiled removal of hybernoblium crystals? Who would _ever_ bother doing this instead of using healium or resin?

**Comment by Pickle-Coding** [comment]
Healium only refills air, which could have the functionality to fix the temperature a little, but isn't very helpful if there's just a lot of cold or hot gas.

The hypernoblium crystal is extremely robust at regulating the temperature to room temperature. It has a pretty large range, and very high heat capacity. Fixing the temperature of a room is probably one of the slower processes in atmospherics.

While atmos resin is pretty OP in a lot of aspects, you'd still have to break the resin so you can move around a room, as well as needing a large water tank if you want to fix multiple rooms.

I want to restrict atmos protection to internals and special suits such as EVA or modsuits, or a few mechs, or any similar gear that is designed to protect you from the hazard. Modsuits, EVA gear, and mechs all have the downside of slowing you down, and requiring power to regulate temperature.

Pressure proofing crystals is powercreep. There is absoltely no reason to not apply it to your clothes, and will grant you very strong protection in a low pressure environment, without any downsides. This basically removes a mechanic for a player, because they no longer have to make the decision to wear gear to protect them from the environment, and can go to low pressure areas as if pressure doesn't matter.

I want to make atmospherics progress less about buffing yourself so atmospheric issues don't matter, and more of allowing you more ways to fix atmospheric issues with new equipment.

**Comment by Mooshimi** [comment]
> Healium only refills air, which could have the functionality to fix the temperature a little, but isn't very helpful if there's just a lot of cold or hot gas.
> 
> The hypernoblium crystal is extremely robust at regulating the temperature to room temperature. It has a pretty large range, and very high heat capacity. Fixing the temperature of a room is probably one of the slower processes in atmospherics.
> 
> While atmos resin is pretty OP in a lot of aspects, you'd still have to break the resin so you can move around a room, as well as needing a large water tank if you want to fix multiple rooms.
> 
> I want to restrict atmos protection to internals and special suits such as EVA or modsuits, or a few mechs, or any similar gear that is designed to protect you from the hazard. Modsuits, EVA gear, and mechs all have the downside of slowing you down, and requiring power to regulate temperature.
> 
> Pressure proofing crystals is powercreep. There is absoltely no reason to not apply it to your clothes, and will grant you very strong protection in a low pressure environment, without any downsides. This basically removes a mechanic for a player, because they no longer have to make the decision to wear gear to protect them from the environment, and can go to low pressure areas as if pressure doesn't matter.
> 
> I want to make atmospherics progress less about buffing yourself so atmospheric issues don't matter, and more of allowing you more ways to fix atmospheric issues with new equipment.

tldr: pr is "do your job atmosians, or we'll make you!"

**Comment by GuillaumePrata** [comment]
> why not remove xenobio fireproof potions, or even the pressureproof potions you can still make in xenobio

I'm pretty sure that is on the choping block too as both are badly designed features.

> remove all roundstart spacesuits because it lets you permanently ignore a part of atmospherics. remove roundstart oxygen, it lets you ignore a part of atmospherics.

HAHAHAHAHAHAHAAHAHAHAHAHAHAHAAHAHAHAHAHAHAH

Amazing argument, thank you for making my day slightly better after reading this.

**Comment by D4C-420** [comment]
> Healium only refills air, which could have the functionality to fix the temperature a little, but isn't very helpful if there's just a lot of cold or hot gas.
proto nitrate is the crystal that does this(refilling the air). healuim actually fixes the air temp wise. 
This is adding a more boring "fix the air via grenade" way that's simply uninteresting given its about the 5th way to do the same result. 

Also why the change is recipe. those numbers look yucky. Just round them.

**Comment by Pickle-Coding** [comment]
Coder sprites added for the grenade active state.

**Comment by Pickle-Coding** [comment]
> > Healium only refills air, which could have the functionality to fix the temperature a little, but isn't very helpful if there's just a lot of cold or hot gas.
> 
> 
> 
> proto nitrate is the crystal that does this(refilling the air). healuim actually fixes the air temp wise. 
> 
> This is adding a more boring "fix the air via grenade" way that's simply uninteresting given its about the 5th way to do the same result. 

Actually healium just fills the air with the base air mix. Basically the same as proto-nitrate, but I don't think it's meant to be able to over pressurize, but it does. The gas it spawns is room temperature, but that is not very efficient and can take many healium crystals (maybe hundreds depending on room size, then you get the problem of having too much gas) to heat a room from 2.7 Kelvin. The hypernoblium foam is far far more powerful than the healium and proto-nitrate crystals when it comes to fixing the air's temperature. The foam has 1.755MJ of heat capacity at room temperature, and also helps accelerate floor air conductivity to bring it to a thermal equilibrium. It just takes a single hypernoblium grenade to heat most rooms from 2.7K to a habitable temperature, and is also very potent for cooling. The healium grenade only spews air effecting up to 7 tiles, only having a heat capacity of 14.56kJ, it's really lame if you try to use it to fix air temperature, and you'd be disappointed.

> Also why the change is recipe. those numbers look yucky. Just round them. 

I raised it, so I have an excuse to make the foam have an even higher heat capacity by assuming hypernoblium's specific heat increases even further when it's in its crystal or foam state.

**Comment by D4C-420** [comment]
> Actually healium just fills the air with the base air mix. Basically the same as proto-nitrate, but I don't think it's meant to be able to over pressurize, but it does. The gas it spawns is room temperature, but that is not very efficient and can take many healium crystals (maybe hundreds depending on room size, then you get the problem of having too much gas) to heat a room from 2.7 Kelvin.

> The hypernoblium foam is far far more powerful than the healium and proto-nitrate crystals when it comes to fixing the air's temperature. The foam has 1.755MJ of heat capacity at room temperature, and also helps accelerate floor air conductivity to bring it to a thermal equilibrium.

> It just takes a single hypernoblium grenade to heat most rooms from 2.7K to a habitable temperature, and is also very potent for cooling. The healium grenade only spews air effecting up to 7 tiles, only having a heat capacity of 14.56kJ, it's really lame if you try to use it to fix air temperature, and you'd be disappointed.

So what your saying is you have decided to make the healuim crystal pointless by making a new one which does its goal but better in everyway? Reads a bit like power creep to me. Also you still have not explained how in the least  how this makes fixing air anymore interesting than using a resin launcher than having the fun of breaking the resin.

**Comment by Pickle-Coding** [comment]
> > Actually healium just fills the air with the base air mix. Basically the same as proto-nitrate, but I don't think it's meant to be able to over pressurize, but it does. The gas it spawns is room temperature, but that is not very efficient and can take many healium crystals (maybe hundreds depending on room size, then you get the problem of having too much gas) to heat a room from 2.7 Kelvin.
> 
> 
> 
> > The hypernoblium foam is far far more powerful than the healium and proto-nitrate crystals when it comes to fixing the air's temperature. The foam has 1.755MJ of heat capacity at room temperature, and also helps accelerate floor air conductivity to bring it to a thermal equilibrium.
> 
> 
> 
> > It just takes a single hypernoblium grenade to heat most rooms from 2.7K to a habitable temperature, and is also very potent for cooling. The healium grenade only spews air effecting up to 7 tiles, only having a heat capacity of 14.56kJ, it's really lame if you try to use it to fix air temperature, and you'd be disappointed.
> 
> 
> 
> So what your saying is you have decided to make the healuim crystal pointless by making a new one which does its goal but better in everyway? Reads a bit like power creep to me. Also you still have not explained how in the least  how this makes fixing air anymore interesting than using a resin launcher than having the fun of breaking the resin. 
> 
> 

The healium crystal refills air. The hypernoblium crystal regulates air temperature. They have different purposes, strengths and weaknesses when it comes to fixing atmospherics.

The resin launcher is better at suffocating fires, and does a good job at fixing temperature, but becomes less practical for larger areas in terms of fixing the temperature. The hypernoblium foam is powerful as it lets air move through it, allowing it to fix temperature outside of its foam.

Some of these crystals could get changed to add new features and upgrades for atmospheric equipment such as the resin launcher, but that's out of the scope of this PR. The purpose of this PR is to change the purpose of the hypernoblium crystal.

**Comment by SylvaraWasTaken** [comment]
> The healium crystal refills air. The hypernoblium crystal regulates air temperature. They have different purposes, strengths and weaknesses when it comes to fixing atmospherics.

The Healium crystal's _intended_ purpose is to set air to the correct temperature; refilling the air is just a bonus side-effect to make it worth its particularly high cost. Hypernoblium costs effectively nothing in comparison, and this outright replaces half of its viability, the much more important half, since a room can be vented and refilled easily with a handful of other methods.

> The resin launcher is better at suffocating fires, and does a good job at fixing temperature, but becomes less practical for larger areas in terms of fixing the temperature. The hypernoblium foam is powerful as it lets air move through it, allowing it to fix temperature outside of its foam.

So it's just objectively better resin. It removes all fire on the tiles it is present on for the duration of the foam, it forces the air to a no-longer-autoignition level preventing it from burning after it fades, it covers a larger area, and it can be moved through. _That_ sounds just as much like powercreep as being able to spaceproof a shirt in a job where you have access to roundstart spacesuits is. I'm not strongly against it, considering it's a fair powercreep since resin is free and roundstart and hypernoblium is not, but it's still powercreep.

I would honestly be all for this change if it was in some way novel and interesting, and i do actually like where you're going with this, I just don't think adding a single crystal that does _ALL_ of atmos' jobs in one click, and invalidates every other tool by merit of its existence, is the right idea. If the crystal were to, for example, temporarily halt all chemical and atmospheric reactions within an area for a duration, bombs included, I would be WAY more interested in this PR. It would be a new and effective tool that is more in line with hyper-noblium's core purpose as a gas.

**Comment by TheBoondock** [comment]
I'm concerned with how hard it is to manufacture if you make it too hard player will be discouraged to make it or are fucked over since they did not spend the entire round preparing. But the temperature regulation part is cool!

**Comment by Pickle-Coding** [comment]
> I'm concerned with how hard it is to manufacture if you make it too hard player will be discouraged to make it or are fucked over since they did not spend the entire round preparing. But the temperature regulation part is cool!

It takes about 30 minutes, but I agree that is a bit much for something that isn't needed every round. I suppose the harder gases that are required for these sorts of things could have some fun uses that doesn't require a hazard or anything, but at the same time I don't want atmos to be completely hermit.

Maybe the crystals could be grinded for liquid hypernoblium, which could calm someone down if they drink it, or used for chemical reactions, or to assist botany. This could be expanded in future PRs, but I don't want to do too much in a single PR.

**Comment by RandomGamer123** [comment]
A reminder to some of the commenters here that #68325 is still pending, and both of these PRs take quite different directions in terms of how easy fixing atmospherics should be, so I would consider both of these PRs to be conflicting just because the new hypernoblium crystal here would be entirely redundant if #68325 were merged

**Comment by Pickle-Coding** [comment]
The cleansing crystal just cleans bad gas and refills air I am assuming. The hypernoblium crystal would still do a better job at fixing the temperature I think.

**Comment by LemonInTheDark** [comment]
(I use the atmospherics tag to mark structural changes that I think are important for past/future understanding of the system and reasoning. 🧹)
oh and I will be reviewing this. I'm not inherently for or against, though I do agree that the current state of the item is pain. I wanted it to be temporary, after removing the xeno item, but the code for both is just horrible and I couldn't figure out how to accomplish that in a stable way.

**Comment by Pickle-Coding** [comment]
I have no idea how to resolve icon conflicts.

**Comment by LemonInTheDark** [comment]
do you need help resolving icon conflicts?

**Comment by LemonInTheDark** [comment]
I'm in theory ok with this change I think, I like the use of foam at least.
it's a bit weird to tie it to hypernob, why do that in particular? if I had to choose between this and JUST the canister freezing for the gas i'd choose #68447, but I don't really think that line needs to be drawn

**Comment by Pickle-Coding** [comment]
Hypernoblium has by far the highest specific heat, so it makes sense for that to be the substance used for fixing thermal problems. I'll get back to it soon and fix some things.

**Comment by Pickle-Coding** [comment]
I'm going to close this as I have higher priorities. I'll take a different approach with this when I open a new PR for this, maybe giving it functionality where people don't feel like they wasted their time if they never needed to use the crystals, without being a straight up upgrade to a person such as pressure resistance.
